kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/mwp/MwpTileProvider.java

```diff
@@ -0,0 +1,195 @@
+    if (mwpTile != null) {
+      return translateTile(mwpTile);
+    }
+  
+    return null;
+  }
+
+  @Override
+  public List<Attachment> listOrganizationTileImages(OrganizationId organizationId, TileId tileId) {
+    fi.otavanopisto.mwp.client.model.Tile mwpTile = findTileByTileId(tileId);
+    if (mwpTile != null) {
+      Integer featuredMediaId = mwpTile.getFeaturedMedia();
+      if (featuredMediaId != null) {
+        fi.otavanopisto.mwp.client.model.Attachment featuredMedia = findMedia(featuredMediaId);
+        if ((featuredMedia != null) && (featuredMedia.getMediaType() == MediaTypeEnum.IMAGE)) {
+          return Collections.singletonList(translateAttachment(featuredMedia));
+        }
+      }
+    }
+  
+    return Collections.emptyList();
+  }
+
+  @Override
+  public Attachment findTileImage(OrganizationId organizationId, TileId tileId, AttachmentId attachmentId) {
+    fi.otavanopisto.mwp.client.model.Tile tile = findTileByTileId(tileId);
+    if (tile != null) {
+      Integer featuredMediaId = tile.getFeaturedMedia();
+      if (featuredMediaId != null) {
+        AttachmentId mwpAttachmentId = getImageAttachmentId(featuredMediaId);
+        if (!idController.idsEqual(attachmentId, mwpAttachmentId)) {
+          return null;
+        }
+        
+        fi.otavanopisto.mwp.client.model.Attachment attachment = findMedia(featuredMediaId);
+        if (attachment != null) {
+          return translateAttachment(attachment);
+        }
+      }
+    }
+  
+    return null;
+  }
+
+  @Override
+  public AttachmentData getTileImageData(OrganizationId organizationId, TileId tileId, AttachmentId attachmentId,
+      Integer size) {
+    
+    Integer mediaId = getMediaId(attachmentId);
+    if (mediaId == null) {
+      return null;
+    }
+    
+    fi.otavanopisto.mwp.client.model.Attachment featuredMedia = findMedia(mediaId);
+    if (featuredMedia.getMediaType() == MediaTypeEnum.IMAGE) {
+      AttachmentData imageData = getImageData(featuredMedia.getSourceUrl());
+      
+      if (size != null) {
+        return scaleImage(imageData, size);
+      } else {
+        return imageData;
+      }
+      
+    }
+    
+    return null;
+  }
+
+  private fi.otavanopisto.mwp.client.model.Tile findTileByTileId(TileId tileId) {
+    TileId kuntaApiId = idController.translateTileId(tileId, MwpConsts.IDENTIFIER_NAME);
+    if (kuntaApiId == null) {
+      logger.severe(String.format("Failed to convert %s into MWP id", tileId.toString()));
+      return null;
+    }
+    
+    ApiResponse<fi.otavanopisto.mwp.client.model.Tile> response = mwpApi.getApi().wpV2TileIdGet(kuntaApiId.getId(), null);
+    if (!response.isOk()) {
+      logger.severe(String.format("Finding tile failed on [%d] %s", response.getStatus(), response.getMessage()));
+    } else {
+      return response.getResponse();
+    }
+    
+    return null;
+  }
+
+  private List<Tile> translateTiles(List<fi.otavanopisto.mwp.client.model.Tile> mwpTiles) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateTiles" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)